File: 刷题/cpp/1773. 统计匹配检索规则的物品数量.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int ans=0;
        int ind=0;
        if(ruleKey=="type") ind=0;
        if(ruleKey=="color") ind=1;
        if(ruleKey=="name") ind=2;
        for(int i=0;i<items.size();i++){
            if(items[i][ind]==ruleValue) ans++;
        }
        return ans;
    }
};

File: 刷题/golang/1773. 统计匹配检索规则的物品数量.go
func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    var d=map[string]int{"type":0,"color":1,"name":2}
    ans:=0
    ind:=d[ruleKey]
    for _,item:=range items{
        if item[ind]==ruleValue {
            ans++
        }
    }
    return ans
}